### ISSUE TITLE
Unable to install

### ISSUE THREAD

**Issue by bernice-rawpixel** [kimai/kimai#1001]
#### Actual behaviour
Trying to install on server. I have it uploaded in this directory: http://rp2-hosting.net/kimai. However, installation error says PHP version is wrong, but upon checking cPanel, we have the correct version.
Memory limit is also sufficient.

ERROR: https://www.screencast.com/t/TJ9kSjPcPqR
PHP VERSION: https://www.screencast.com/t/9Y8sFDKdu0ei

**Comment by simonschaufi** [comment]
Your problem does not seam the php version. The installer returns: `500 Internal Server Error` in this step.

**Comment by bernice-rawpixel** [comment]
Hi

What does that mean? What do I need to do?

Regards,

Bernice Jennifer Bontalilid
Web Development Team Lead Mobile: +63948.797.5991
Email: bernice@rawpixel.com.au
Skype: rawpixel.bernice

**Comment by ToeiRei** [comment]
can you get some more debug log? Looks like something (on the webserver?) is misconfigured.

**Comment by Mistralys** [comment]
I had the same issue. The requirements screen showed the PHP version as insufficient.

Checking the requests in the page, I saw that the call to the `processor.php` file throws a PHP notice. After investigating, the culprit is the function `return_bytes`.

To fix it, I did the following:

1. Open the file installer/processor.php
2. Change the PHP code for the return_bytes function in line 30 to the following:

```php
function return_bytes($val) {
    $val = trim($val);
    $last = $val[strlen($val) - 1];
    $val = rtrim($val, $last);
    switch (strtolower($last)) {
        // The 'G' modifier is available since PHP 5.1.0
        case 'g':
            $val *= 1073741824;
        case 'm':
            $val *= 1048576;
        case 'k':
            $val *= 1024;
    }

    return $val;
}
```
The method did not remove the units character from the value (e.g. 100M), so trying to multiply that string with a value throws a notice and does not return the expected amount of bytes.
